specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp

```diff
@@ -1185,38 +1185,59 @@ model AcsMessageDeliveryStatusUpdatedEventData extends AcsMessageEventData {
 /** Schema of the Data property of an EventGridEvent for a Microsoft.Communication.AdvancedMessageReceived event. */
 #suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "Maintain compatibility with existing models."
 model AcsMessageReceivedEventData extends AcsMessageEventData {
-  /** The message content */
+  /** Optional. The message content */
   content?: string;
 
-  /** The message channel type */
+  /** Optional. Message ID. Format is Guid as string.  */
+  messageId?: string;
+
+  /** Required. The message channel type */
   channelType: AcsMessageChannelKind;
 
-  /** The received message media content */
-  media: AcsMessageMediaContent;
+  /** Required. Whatsapp message type */
+  messageType: string;
+
+  /** Optional. The received message media content */
+  media?: AcsMessageMediaContent;
 
-  /** The received message context */
-  context: AcsMessageContext;
+  /** Optional. The received message reaction content */
+  reaction?: AcsMessageReactionContent;
 
-  /** The received message button content */
-  button: AcsMessageButtonContent;
+  /** Optional. The received message context */
+  context?: AcsMessageContext;
 
-  /** The received message interactive content */
-  interactive: AcsMessageInteractiveContent;
+  /** Optional. The received message button content */
+  button?: AcsMessageButtonContent;
+
+  /** Optional. The received message interactive content */
+  interactive?: AcsMessageInteractiveContent;
 }
 
 /** Message Media Content */
 model AcsMessageMediaContent {
-  /** The MIME type of the file this media represents */
-  mimeType?: string;
+  /** Required. The MIME type of the file this media represents */
+  mimeType: string;
 
-  /** The media identifier */
-  id?: string;
+  /** Required. The media identifier */
+  id: string;
 
-  /** The filename of the underlying media file as specified when uploaded */
+  /** Optional. The filename of the underlying media file as specified when uploaded */
   fileName?: string;
 
-  /** The caption for the media object, if supported and provided */
+  /** Optional. The caption for the media object, if supported and provided */
   caption?: string;
+
+  /** Optional. Set to  true  if the sticker is animated; false otherwise. */
+  animated?: boolean;
+}
+
+/** Message Reaction Content */
+model AcsMessageReactionContent {
+  /** Required. WhatsApp message ID of message you want to apply the emoji to. */
```

Comment:
Doesn't this represent an event that already occurred? I was expecting the description to say something like "WhatsApp message ID of the message that the emoji is applied to.